Lib/inspect.py

```diff
@@ -2495,6 +2477,46 @@ def _signature_from_function(cls, func, skip_bound_arg=True,
                __validate_parameters__=is_duck_function)
 
 
+def _signature_from_builtin(cls, func, skip_bound_arg=True):
+    """Private helper function to get signature for
+    builtin callables.
+    """
+
+    if not _signature_is_builtin(func):
+        raise TypeError("{!r} is not a Python builtin "
+                        "function".format(func))
```

Comment:
Private functions don't have to check their arguments, you already do the exact same check before calling the function. If users use private functions, they are on their own.